src/libraries/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.cs

```diff
@@ -158,6 +158,12 @@ public override object Invoke(BindingFlags invokeAttr, Binder? binder, object?[]
                 throw new TargetParameterCountException(SR.Arg_ParmCnt);
             }
 
+            // Array construction is a special case.
+            if (DeclaringType is not null && DeclaringType.IsArray)
+            {
+                return InvokeArrayCtorWorker(parameters!);
```

Comment:
We will skip `CheckArguments` below for the array constructor now. Is `CheckArguments` guaranteed to be no-op for the array constructors? It is not obvious to me that it is the case.

Reply:
I believe it is based on what is expected in the new `InvokeArrayCtor` which was previously `InvokeArrayConstructor` and has the following argument processing&mdash;all arguments must be primitives and "widenable" to `I4`. I don't see the logic in `CheckArguments` that would needed to run for this case.

```cpp
        if (!objs->GetAt(i))
            COMPlusThrowArgumentException(W("parameters"), W("Arg_NullIndex"));

        MethodTable* pMT = objs->GetAt(i)->GetMethodTable();
        CorElementType oType = TypeHandle(pMT).GetVerifierCorElementType();

        if (!InvokeUtil::IsPrimitiveType(oType) || !InvokeUtil::CanPrimitiveWiden(ELEMENT_TYPE_I4,oType))
            COMPlusThrow(kArgumentException,W("Arg_PrimWiden"));
```

Reply:
This prints "2" today. Is it still work after this change?
```csharp
using System.Globalization;
using System.Reflection;

var ctor = typeof(int[,]).GetConstructor(new[] { typeof(int), typeof(int) });
var arr = ctor.Invoke(BindingFlags.Default, new MyBinder(), new object[] { "1", "2" }, null);
Console.WriteLine(((int[,])arr).Length);

class MyBinder : Binder
{
    public override FieldInfo BindToField(BindingFlags bindingAttr, FieldInfo[] match, object value, CultureInfo? culture)
        => throw new NotImplementedException();

    public override MethodBase BindToMethod(BindingFlags bindingAttr, MethodBase[] match, ref object?[] args, ParameterModifier[]? modifiers, CultureInfo? culture, string[]? names, out object? state)
        => throw new NotImplementedException();

    public override object ChangeType(object value, Type type, CultureInfo? culture)
        => int.Parse((string)value);

    public override void ReorderArgumentArray(ref object?[] args, object state)
        => throw new NotImplementedException();

    public override MethodBase? SelectMethod(BindingFlags bindingAttr, MethodBase[] match, Type[] types, ParameterModifier[]? modifiers)
        => throw new NotImplementedException();

    public override PropertyInfo? SelectProperty(BindingFlags bindingAttr, PropertyInfo[] match, Type? returnType, Type[]? indexes, ParameterModifier[]? modifiers)
        => throw new NotImplementedException();
}
```

Reply:
```csharp
        public override object ChangeType(object value, Type type, CultureInfo? culture)
            => int.Parse((string)value);
```

-.-    That is evil...

Reply:
I am adding a test case too.